lib/request_overrider.js

```diff
@@ -1,604 +0,0 @@
-  // https://nodejs.org/api/http.html#http_response_socket
-  // https://github.com/nodejs/node/blob/b0f75818f39ed4e6bd80eb7c4010c1daf5823ef7/lib/_http_incoming.js#L44-L69
-  // The same Socket is shared between the request and response to mimic native behavior.
-  req.socket = req.connection = new Socket({ proto: options.proto })
-  response.socket = response.client = response.connection = req.socket
-
-  propagate(['timeout'], req.socket, req)
-
-  // Emit a fake socket event on the next tick to mimic what would happen on a real request.
-  // Some clients listen for a 'socket' event to be emitted before calling end(),
-  // which causes nock to hang.
-  process.nextTick(() => {
-    req.emit('socket', req.socket)
-
-    // https://nodejs.org/api/net.html#net_event_connect
-    req.socket.emit('connect')
-
-    // https://nodejs.org/api/tls.html#tls_event_secureconnect
-    if (req.socket.authorized) {
-      req.socket.emit('secureConnect')
-    }
-  })
-
-  req.write = function(buffer, encoding, callback) {
-    debug('write', arguments)
-    if (!req.aborted) {
-      if (buffer) {
-        if (!Buffer.isBuffer(buffer)) {
-          buffer = Buffer.from(buffer, encoding)
-        }
-        requestBodyBuffers.push(buffer)
-      }
-      if (typeof callback === 'function') {
-        callback()
-      }
-    } else {
-      emitError(new Error('Request aborted'))
-    }
-
-    timers.setImmediate(function() {
-      req.emit('drain')
-    })
-
-    return false
-  }
-
-  req.end = function(chunk, encoding, callback) {
-    debug('req.end')
-    if (typeof chunk === 'function') {
-      callback = chunk
-      chunk = null
-    } else if (typeof encoding === 'function') {
-      callback = encoding
-      encoding = null
-    }
-
-    if (!req.aborted && !ended) {
-      req.write(chunk, encoding, () => {
-        if (typeof callback === 'function') {
-          callback()
-        }
-        end()
-        req.emit('finish')
-        req.emit('end')
-      })
-    }
-    if (req.aborted) {
-      emitError(new Error('Request aborted'))
-    }
-  }
-
-  req.flushHeaders = function() {
-    debug('req.flushHeaders')
-    if (!req.aborted && !ended) {
-      end()
-    }
-    if (req.aborted) {
-      emitError(new Error('Request aborted'))
-    }
-  }
-
-  req.abort = function() {
-    if (req.aborted) {
-      return
-    }
-    debug('req.abort')
-    req.aborted = Date.now()
-    if (!ended) {
-      end()
-    }
-    const err = new Error()
-    err.code = 'aborted'
-    response.emit('close', err)
-
-    req.socket.destroy()
-
-    req.emit('abort')
-
-    const connResetError = new Error('socket hang up')
-    connResetError.code = 'ECONNRESET'
-    emitError(connResetError)
-  }
-
-  const emitError = function(error) {
-    process.nextTick(function() {
-      req.emit('error', error)
-    })
-  }
-
-  const end = function() {
-    debug('ending')
-    ended = true
-    let requestBody, responseBody, responseBuffers
```

Comment:
`requestBody` is now a local variable in `end()`. `{ responseBody, responseBuffers }` become named parameters to `continueWithResponseBody()`.